sklearn/metrics/_plot/roc_curve.py

```diff
@@ -118,11 +142,37 @@ def plot(self, ax=None, *, name=None, **kwargs):
 
         line_kwargs.update(**kwargs)
 
+        chance_level_line_kwargs = {
+            "label": "Chance level (AUC = 0.5)",
+            "color": "k",
+            "linestyle": "--",
+        }
+
+        chance_level_line_kwargs.update(**chance_level_kwargs)
+
         import matplotlib.pyplot as plt
 
         if ax is None:
             fig, ax = plt.subplots()
 
+        # Make sure that the chance level line is not plotted multiple times
```

Comment:
Let's keep it simple. This is a bit over-engineered :).
We plot it only if the user requests it. If we plot the data twice, we might just change the colour but this is not a big deal. The user will be able to update the code if the documentation and example are showing how to use the feature.